protocol/messenger_communities.go

```diff
@@ -706,6 +706,21 @@ func (m *Messenger) RequestToJoinCommunity(request *requests.RequestToJoinCommun
 
 		for _, walletAccount := range walletAccounts {
 			if !walletAccount.Chat && walletAccount.Type != accounts.AccountTypeWatch {
+				if len(request.Addresses) > 0 {
+					// only use the addresses the user wants to share
+					found := false
+					for _, address := range request.Addresses {
+						if address == walletAccount.Address.Hex() {
+							found = true
+							break
+						}
+					}
+					if !found {
+						// The user didn't select this address
+						continue
+					}
+				}
+
```

Comment:
Just another way to do it, but would be cool if we could change the check to something like

```golang
if wantsToRevealWalletAddress := addressToRevealMap[walletAccount.Address.Hex()]; hasAddressesToReveal && wantsToRevealWalletAddress {
 ...
}
```